### ISSUE TITLE
Unable to run sf plugins link

### ISSUE THREAD

**Issue by nvuillam** [forcedotcom/cli#2818]
Since today, we're unable to locally link my plugin sources to my SF CLI

This makes us **unable to enhance / debug [sfdx-hardis](https://sfdx-hardis.cloudity.com/)**, so it is critical (especially knowing that I have an urgent fix to test & publish)

Is this related to https://github.com/forcedotcom/cli/issues/2691 ?

Note: I checked locally on my Windows computer + on a brand new GitHub CodeSpaces so this is not local-env related

Please help ? :)

To reproduce:

```
git clone https://github.com/hardisgroupcom/sfdx-hardis.git
cd sfdx-hardis
yarn
npm i @salesforce/cli -g
sf plugins link
sf plugins reset --reinstall --hard
sf plugins link
```

Full log

```
@nvuillam ➜ /workspaces/sfdx-hardis (main) $ git clone https://github.com/hardisgroupcom/sfdx-hardis.git
Cloning into 'sfdx-hardis'...
remote: Enumerating objects: 13723, done.
remote: Counting objects: 100% (1565/1565), done.
remote: Compressing objects: 100% (674/674), done.
remote: Total 13723 (delta 899), reused 1259 (delta 668), pack-reused 12158
Receiving objects: 100% (13723/13723), 30.24 MiB | 26.88 MiB/s, done.
Resolving deltas: 100% (8429/8429), done.
@nvuillam ➜ /workspaces/sfdx-hardis (main) $ cd sfdx-hardis
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ yarn
yarn install v1.22.19
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
[5/5] Building fresh packages...
Done in 52.87s.
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ npm i @salesforce/cli -g

added 953 packages in 36s

122 packages are looking for funding
  run `npm fund` for details
npm notice 
npm notice New patch version of npm available! 10.5.0 -> 10.5.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.2
npm notice Run npm install -g npm@10.5.2 to update!
npm notice 
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ sf plugins link
@salesforce/cli: Linking plugin .... !
 ›   Error: /usr/local/share/nvm/versions/node/v20.12.1/lib/node_modules/@salesforce/cli/node_modules/@oclif/plugin-plugins/node_modules/npm/bin/npm-cli.js 
 ›   install --no-audit --loglevel=silent --no-fund exited with code 1
 ›   Try this: Run with DEBUG=@oclif/plugin-plugins* to see debug output.
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ sf plugins reset --reinstall --hard
Found 0 plugin:
✅ Removed the following files:
- /home/codespace/.local/share/sf/node_modules
- /home/codespace/.local/share/sf/package.json
- /home/codespace/.local/share/sf/yarn.lock
- /home/codespace/.local/share/sf/package-lock.json
Reinstall flag passed:
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ sf plugins link
@salesforce/cli: Linking plugin .... !
 ›   Error: /usr/local/share/nvm/versions/node/v20.12.1/lib/node_modules/@salesforce/cli/node_modules/@oclif/plugin-plugins/node_modules/npm/bin/npm-cli.js 
 ›   install --no-audit --loglevel=silent --no-fund exited with code 217
 ›   Try this: Run with DEBUG=@oclif/plugin-plugins* to see debug output.
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ 
```

**Comment by nvuillam** [comment]
CLI info 

```
@nvuillam ➜ /workspaces/sfdx-hardis/sfdx-hardis (main) $ sf version --verbose --json
{
  "architecture": "linux-x64",
  "cliVersion": "@salesforce/cli/2.36.8",
  "nodeVersion": "node-v20.12.1",
  "osVersion": "Linux 6.2.0-1019-azure",
  "rootPath": "/usr/local/share/nvm/versions/node/v20.12.1/lib/node_modules/@salesforce/cli",
  "shell": "bash",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.13 (core)",
    "@oclif/plugin-commands 3.2.2 (core)",
    "@oclif/plugin-help 6.0.20 (core)",
    "@oclif/plugin-not-found 3.1.1 (core)",
    "@oclif/plugin-plugins 5.0.5 (core)",
    "@oclif/plugin-search 1.0.20 (core)",
    "@oclif/plugin-update 4.2.3 (core)",
    "@oclif/plugin-version 2.0.16 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.7 (core)",
    "@salesforce/cli 2.36.8 (core)",
    "apex 3.1.3 (core)",
    "auth 3.5.1 (core)",
    "data 3.2.4 (core)",
    "deploy-retrieve 3.5.3 (core)",
    "info 3.1.2 (core)",
    "limits 3.2.1 (core)",
    "marketplace 1.1.0 (core)",
    "org 3.6.3 (core)",
    "packaging 2.3.0 (core)",
    "schema 3.2.0 (core)",
    "settings 2.1.2 (core)",
    "sobject 1.2.1 (core)",
    "source 3.2.2 (core)",
    "telemetry 3.1.17 (core)",
    "templates 56.1.0 (core)",
    "trust 3.5.1 (core)",
    "user 3.4.3 (core)"
  ]
}
```

**Comment by mdonnalley** [comment]
Nicolas and I talked about this in slack, but for everyone else who might encountering this:

We have a PR to plugin-plugins to fix this https://github.com/oclif/plugin-plugins/pull/841

In the meantime, the workaround is to use the `--no-install` flag on `sf plugins link` so that the install step is skipped entirely. If you do this, then you must ensure that you've installed your dependencies prior to the link (e.g. run `yarn` inside your plugin directory)

**Comment by iowillhoit** [comment]
Hey @nvuillam, this has been merged. It'll be in tonight's `nightly`, in `latest-rc` next Wednesday, and in `latest` on May 1st.

**Comment by nvuillam** [comment]
> Hey @nvuillam, this has been merged. It'll be in tonight's `nightly`, in `latest-rc` next Wednesday, and in `latest` on May 1st. 

Many thanks :)

**Comment by jshackell-sfdc** [comment]
This issue is fixed in version 2.39.6 of Salesforce CLI.
